src/db/models.py

```diff
@@ -1,48 +1,101 @@
 from . import connection
 import psycopg2.sql as sql
+import abc
 
-class ApplicationsModel:
-    TableName = "applications"
+class BaseModel(abc.ABC):
+    TableName = ""
     Model = {}
 
     @classmethod
-    def setModel(cls, fieldnames, sample_data):
-        for field in fieldnames:
-            cls.Model[field] = str # force everything into string, deal with on client
-        
-        cls.Model["score"] = int
-        
+    def dropTable(cls):
+        conn = connection.get_connection()
+        cursor = conn.cursor()
+        try:
+            cursor.execute('DROP TABLE IF EXISTS "%s"' % cls.TableName)
+            conn.commit()
+        except Exception as e:
+            raise(e)
+        finally:
+            cursor.close()
+            conn.close()
+
+    @classmethod
+    def createTable(cls):
         conn = connection.get_connection()
         cursor = conn.cursor()
         try:
-            cursor.execute('DROP TABLE IF EXISTS %s' % ApplicationsModel.TableName)
-            cursor.execute('CREATE TABLE %s ()' % ApplicationsModel.TableName)
+            cursor.execute('CREATE TABLE "%s" ()' % cls.TableName)
             
             for name, data_type in cls.Model.items():
+                if type(data_type) is str:
+                    postgre_type = data_type
                 if data_type is str:
                     postgre_type = "TEXT"
                 if data_type is int:
                     postgre_type = "SMALLINT"
                 if data_type is float:
                     postgre_type = "real"
-                cursor.execute('ALTER TABLE %s ADD COLUMN "%s" %s' % (cls.TableName, name, postgre_type))
+                cursor.execute('ALTER TABLE "%s" ADD COLUMN "%s" %s' % (cls.TableName, name, postgre_type))
 
             conn.commit()
         except Exception as e:
-            print(e)
-        finally:    
+            conn.rollback()
+            raise(e)
+        finally:
             cursor.close()
             conn.close()
 
 
+class UserModel(BaseModel):
+    TableName = "user"
+    Model = {}
+
+    @classmethod
+    def createUserTable(cls):
+        cls.Model['id'] = int
```

Comment:
What if we move this down to after `cls.createTable` so that we don't have to drop it later.

Reply:
makes sense